### PR TITLE
Sandre/api video creation

### PR DESCRIPTION
**Pull request by sandre35**

### CODE REVIEW

File: tournesol/serializers.py [hunk 1]
```diff
@@ -13,6 +13,12 @@ class Meta:
         fields = ["video_id"]
 
 
+class VideoCriteriaScoreSerializer(ModelSerializer):
+    class Meta:
+        model = VideoCriteriaScore
+        fields = "__all__"
```
Comment (elineda) [inline tournesol/serializers.py:19]: c'est sur que l'on met tout ?

File: tournesol/views/video.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+class VideoList(APIView):
+
+    permission_classes = []  # To unlock authentification required
+
+    def get(self, *args, **kwargs):
+        "Get video details and criteria that are related to it"
+        try:
+            video = Video.objects.get(video_id=self.kwargs["youtube_video_id"])
+        except Video.DoesNotExist:
```
Comment (elineda) [inline tournesol/views/video.py:21]: don't raise postgres error. do get object or 404

File: tournesol/views/video.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                "The video doesn't exist",
+                status=status.HTTP_404_NOT_FOUND
+            )
+        video_criterias = VideoCriteriaScore.objects.filter(
+            video=video
+        )
+        video_serializered = VideoSerializer(video)
+        criteria_dict = {}
+        for criteria in video_criterias:
+            criteria_dict[criteria.criteria] = VideoCriteriaScoreSerializer(criteria)
```
Comment (elineda) [inline tournesol/views/video.py:32]: this should be on serialiser no there

File: tournesol/views/video.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        return Response((video_serializered.data, criteria_dict), status=status.HTTP_200_OK)
+
+    def post(self, request):
+        """
+        Add a video to the db if it does not already exist
+        """
+
+        serializer = VideoSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
```
Comment (elineda) [inline tournesol/views/video.py:42]: where do you save VideoCriteriaScore ?

**Review by elineda — CHANGES_REQUESTED** [review]

File: tournesol/views/video.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+from rest_framework.views import APIView
+from rest_framework import status
+from rest_framework.response import Response
+
+from tournesol.serializers import VideoSerializer, VideoCriteriaScoreSerializer
+from tournesol.models import VideoCriteriaScore, Video
+
+
+class VideoList(APIView):
```
Comment (elineda) [inline tournesol/views/video.py:13]: why no use modelviewset ?

**Review by elineda — CHANGES_REQUESTED** [review]

File: tournesol/views/video.py (hunk 4 above)
Comment (lfaucon) [reply tournesol/views/video.py:42]: VideoCriteriaScores are set by the ML algorithm on the backend. This API endpoint should not edit them

File: tournesol/views/video.py (hunk 5 above)
Comment (sandre35) [reply tournesol/views/video.py:13]: @elineda This is what is recommended in the documentation : https://www.django-rest-framework.org/tutorial/3-class-based-views/

File: tournesol/views/video.py (hunk 4 above)
Comment (sandre35) [reply tournesol/views/video.py:42]: What is saved is the Video, not the VideoCriteriaScore. I call the VideoSerializer, not VideoCriteriaScoreSerializer

File: tournesol/views/video.py (hunk 5 above)
Comment (elineda) [reply tournesol/views/video.py:13]: yeah but no... part 4 of tutorial 👍 https://www.django-rest-framework.org/tutorial/4-authentication-and-permissions/
look what they use as class

File: tournesol/tests/test.video.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+from django.test import TestCase
```
Comment (lfaucon) [inline tournesol/tests/test.video.py:1]: the file `test.video.py` is not selected by pytest when running tests (neither on the ci, neither when running `pytest tournesol` locally). Fixing the name to `test_video.py` of `test_api_video.py` works well with pytest. 

However the tests fail currently.

Additionally I'd recommend adding a test with incorrect video ids. And adding two tests for the behavior when fetching an existing or non-existing video

**Review by lfaucon — CHANGES_REQUESTED** [review]

File: tournesol/urls.py [hunk 7]
```diff
@@ -29,4 +30,9 @@
         VideoRateLaterDetail.as_view(),
         name="video_rate_later_detail",
     ),
+    path(
+        "videos/<str:youtube_video_id>",
```
Comment (lfaucon) [inline tournesol/urls.py:34]: I'm not sure about this, but I recall some confusing behaviour with or without a `/` at the end of the url path. Could you check what is recommended for this and either modify the `videos` path or the two above?

File: tournesol/serializers.py [hunk 8]
```diff
@@ -13,6 +14,17 @@ class Meta:
...
+
+
+class VideoAllInformationSerializer(serializers.Serializer):
+    video = VideoSerializer(required=False)
+    videocriterias = VideoCriteriaScoreSerializer(many=True, required=False)
```
Comment (lfaucon) [inline tournesol/serializers.py:25]: I'd recommend renaming this attribute `criterias` for simplicity

**Review by lfaucon — APPROVED** [review]

**Pull request merged**